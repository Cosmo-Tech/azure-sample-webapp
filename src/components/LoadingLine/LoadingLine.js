// Copyright (c) Cosmo Tech.
// Licensed under the MIT license.

import React from 'react';
import PropTypes from 'prop-types';
import * as dataLoading from '../../assets/loadingView/dataLoading.json';
import * as dataLoaded from '../../assets/loadingView/dataLoaded.json';
import * as dataError from '../../assets/loadingView/dataError.json';
import Lottie from 'react-lottie';
import { useTranslation } from 'react-i18next';
import { Grid, Typography } from '@material-ui/core';

const dataLoaderOptions = {
  loop: true,
  autoplay: true,
  animationData: dataLoading.default,
  rendererSettings: {
    preserveAspectRatio: 'xMidYMid slice'
  }
};

const dataLoadedOptions = {
  loop: false,
  autoplay: true,
  animationData: dataLoaded.default,
  rendererSettings: {
    preserveAspectRatio: 'xMidYMid slice'
  }
};
const errorDataOptions = {
  loop: false,
  autoplay: true,
  animationData: dataError.default,
  rendererSettings: {
    preserveAspectRatio: 'xMidYMid slice'
  }
};

const LoadingLine = props => {
  const { t } = useTranslation();
  const { titleKey, isLoading, hasError, height, width } = props;
  return (
      <>
        <Grid container direction="row" alignItems={'center'}>
          <Grid item >
            <Typography variant="h2">
              {t(titleKey, 'LoadingLine Title')}
            </Typography>
          </Grid>
          <Grid item >
            {isLoading
              ? (<Lottie options={dataLoaderOptions} height={height} width={width} />)
              : (<Lottie options={hasError ? errorDataOptions : dataLoadedOptions} height={height} width={width} />)}
          </Grid>
        </Grid>
      </>
  );
};

LoadingLine.propTypes = {
  titleKey: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  hasError: PropTypes.bool.isRequired,
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired
};

export default LoadingLine;
