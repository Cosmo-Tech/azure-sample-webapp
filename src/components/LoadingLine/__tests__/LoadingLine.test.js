// Copyright (c) Cosmo Tech.
// Licensed under the MIT license.

import React from 'react';
import renderer from 'react-test-renderer';
import LoadingLine from '../LoadingLine';

describe('LoadingLine test suite', () => {
  console.error = jest.fn();

  it('should log errors when no props are passed', () => {
    renderer.create(<LoadingLine />).toJSON();
    expect(console.error).toHaveBeenCalledTimes(5);
  });

  it('should render correctly when all props are here', () => {
    const elem = renderer.create(
      <LoadingLine
        titleKey={'test.all.props.key'}
        width={120}
        height={120}
        isLoading={true}
        hasError={false}
      />
    ).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render correctly when its loading', () => {
    const elem = renderer.create(
      <LoadingLine
        titleKey={'test.is.loading.key'}
        width={120}
        height={120}
        isLoading={true}
        hasError={false}
      />
    ).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render correctly when its loaded', () => {
    const elem = renderer.create(
      <LoadingLine
        titleKey={'test.is.loaded.key'}
        width={120}
        height={120}
        isLoading={false}
        hasError={false}
      />
    ).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render correctly when it has errors', () => {
    const elem = renderer.create(
      <LoadingLine
        titleKey={'test.has.errors.key'}
        width={120}
        height={120}
        isLoading={false}
        hasError={true}
      />
    ).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
