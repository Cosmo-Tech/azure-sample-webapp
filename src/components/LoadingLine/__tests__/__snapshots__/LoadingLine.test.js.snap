// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingLine test suite should render correctly when all props are here 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      test.all.props.key
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      aria-label="animation"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "height": "120px",
          "margin": "0 auto",
          "outline": "none",
          "overflow": "hidden",
          "width": "120px",
        }
      }
      tabIndex="0"
      title=""
    />
  </div>
</div>
`;

exports[`LoadingLine test suite should render correctly when it has errors 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      test.has.errors.key
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      aria-label="animation"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "height": "120px",
          "margin": "0 auto",
          "outline": "none",
          "overflow": "hidden",
          "width": "120px",
        }
      }
      tabIndex="0"
      title=""
    />
  </div>
</div>
`;

exports[`LoadingLine test suite should render correctly when its loaded 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      test.is.loaded.key
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      aria-label="animation"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "height": "120px",
          "margin": "0 auto",
          "outline": "none",
          "overflow": "hidden",
          "width": "120px",
        }
      }
      tabIndex="0"
      title=""
    />
  </div>
</div>
`;

exports[`LoadingLine test suite should render correctly when its loading 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      test.is.loading.key
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      aria-label="animation"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "height": "120px",
          "margin": "0 auto",
          "outline": "none",
          "overflow": "hidden",
          "width": "120px",
        }
      }
      tabIndex="0"
      title=""
    />
  </div>
</div>
`;
